### ISSUE TITLE
Feature request - knocking

### ISSUE THREAD

**Issue by lordrak007** [home-assistant/android#3432]
**Is your feature request related to a problem? Please describe.**
Add feature to client, to enable port knocking before client conenct.

**Describe the solution you'd like**
To have advances field under HA url setup where user can enter knock sequence.
Sequence could be for example: 15001:tcp, 0,5s: 10010:tcp, 0,5s: 15000:udp

App send packet to port 15001 using TCP, then wait 0.5 second, send packet to 10010:tcp, wait 0.5 second, sedn UDP packet to port 15000
**Describe alternatives you've considered, if any**
No other alternative

**Additional context**
This could enhance security layer for advance users. Knocking can be setup on many routers and hides port 8123 fo all not knocking users. Hides publis shared home assistent so scanners can see it.

**Comment by dshokouhi** [comment]
I am not sure this is going to benefit the majority of users and it also feels this will greatly confuse a user. I am also not sure if this is something the app should maintain. This request feels like a very big edge case and I do not think HA core supports this.

**Comment by lordrak007** [comment]
I understand. This is not for mainstream but for user who want more secure access to their onrepm HA setup.
This may be hidden for user in standard mode, shows only after user click to show it.
This may be not so hard to develop it.
Please give a chance to this security leayer.
Thank you

**Comment by dshokouhi** [comment]
> I understand. This is not for mainstream but for user who want more secure access to their onrepm HA setup.
> This may be hidden for user in standard mode, shows only after user click to show it.

Hiding features is not good UX also this needs to be something easily explainable to the average user who may find it and enable it and break things.  Yes we have been in cases before where users enabled things without realizing what it is, that is why we want to avoid adding confusing features that may not benefit the majority. Do you have an example of an app that does this today? 



> This may be not so hard to develop it.

Will you be submitting the PR since you are able to judge the amount of work needed?

**Comment by lordrak007** [comment]
I did nt want hide it definetely. Sorry i am not native english speaker so i cant explain some things clearly.
By hideing i mean some drop down menu with advance option.
It could not break nothing because when someone setup it, it only send dew packets which do nothing if you do not know what you want.
I have not example in Android world but in Windows workd yes. Also this is not new in newtwork administrators group.

No, i will not :) I did not write how anmout of work is needed. Only wrote this is not dificult to do. I can do it easy in powershell and .Net but i don't in Android. So i can imagine clearly how diffitult it is.

**Comment by dshokouhi** [comment]
I do not think this is a request we should accept.  Honestly speaking I never heard of this until today.

**Comment by Mincka** [comment]
Port knocking is just security by obscurity. You should clearly look for proper VPN setup to secure the access to your instance.

**Comment by lordrak007** [comment]
It is not security by obscurity but yes, it is not widely used.  VPN can be used also so if you could implement auto dial vpn for application only. It will be best but it is more difficult to implemented then knocking but moch more secureand universal.

**Comment by virtualdj** [comment]
> VPN can be used also so if you could implement auto dial vpn for application only

There's the _TLS Client Authentication_ that works perfectly with Android, without the need of a VPN. Unfortunately, there are still issues with Wear OS.
